### ISSUE TITLE
Provider API error

### ISSUE THREAD

**Issue by abemedia** [scaleway/juju-scaleway#8]
Running `juju scaleway bootstrap` returns the following error:

```
Provider interaction error: <ProviderAPIError message:Image 'cb7d7bb5-54b8-45f4-bee4-164b2aedc69c' arch is not 'arm'. response:400>
```

**Comment by moul** [comment]
@bchatelard, this plugin is probably outdated and does not support multiarch yet, can you give a look and fix it or explain the needed changes ?

**Comment by bchatelard** [comment]
Yeah, i think you should add a `commercial_type` param to server creation dict here https://github.com/scaleway/juju-scaleway/blob/5eefae729591a17f88c3d256bd084ba1a73ffd87/juju_scaleway/client.py#L65-L70

There is a mismatch between the image arch and the server arch (which is C1 by default).

Something like this should do the trick:

```
    def create_server(self, name, image):
        commercial_type = <read from config> or 'VC1'
        params = dict(
            name=name,
            image=image,
            commercial_type=commercial_type,
            organization=self.access_key)

        data = self.request('/servers', method='POST', params=params)
```

**Comment by bchatelard** [comment]
Also a rework is needed for the image selection, the current implementation is based on a text matching over the /images endpoint.

We need to expose the wanted arch to the config, and to query the marketplace api based on the location and the arch.

**Comment by abemedia** [comment]
Does this mean deploying on C2 or VPS will be possible using the `--constraints` flag?
So building the state server on a C2 would be something like the following.

```
juju scaleway bootstrap --constraints "arch=amd64 mem=8G"
```

**Comment by bchatelard** [comment]
Yes, we can implement this using constraints https://jujucharms.com/docs/stable/reference-constraints

**Comment by sheerun** [comment]
Still an issue. Any plans to fix it?
